pkg/scheduler/cache/usagedb/prometheus/prometheus.go

```diff
@@ -59,15 +62,33 @@ func NewPrometheusClient(address string, params *api.UsageParams) (api.Interface
 		"memory":         params.GetExtraStringParamOrDefault("memoryAllocationMetric", "kai_queue_allocated_memory_bytes"),
 	}
 
-	return &PrometheusClient{
+	clientObj := &PrometheusClient{
 		client:      v1api,
 		promClient:  client,
 		usageParams: params,
 
 		usageQueryTimeout:    usageQueryTimeout,
 		queryResolution:      queryResolution,
 		allocationMetricsMap: allocationMetricsMap,
-	}, nil
+	}
+
+	if params.WindowType == nil {
+		return nil, fmt.Errorf("window type is not set in usage params")
+	}
+	switch *params.WindowType {
+	case api.TumblingWindow:
+		clientObj.queryUsageFunction = clientObj.queryTumblingWindowUsage
+
+		cronExpression, err := cronexpr.Parse(params.CronString)
```

Comment:
Cron validation should probably be validated globally, in `pkg/scheduler/cache/usagedb/hub.go`, since it's not a client-specific thing